src/node/construct.rs

```diff
@@ -48,6 +49,42 @@ impl<J: Jet> ConstructNode<J> {
     pub fn arrow(&self) -> &Arrow {
         &self.data.arrow
     }
+
+    /// Sets the source and target type of the node to unit
+    pub fn type_as_program(&self) -> Result<(), types::Error> {
```

Comment:
b3bce4e36dc504598954839cb0c9ab3556a2725e: Confusing method name. Maybe `set_program_type`? At least `set` should be in the name.